### DISCUSSION TITLE
Year embedding

### DISCUSSION THREAD

**Discussion by asger1537** [AIStream-Peelout/flow-forecast#646] [Q&A] [answered]
I tried to make a config with year, month, day, day_of_week as temporal features for an informer model but found out it tries to use  flood_forecast.transformer_xl.data_embedding.TemporalEmbedding which doesn't seem to handle years.  Does the TemporalEmbedding intentionally not deal with years? Is there any other way to do the year embedding or is it completely left out because it very rarely improves the performance of the model? 
Thanks

**Comment by isaacmg** [comment, marked as answer]
At the moment a year embedding is not supported. The original Informer paper did not have a year embedding. I think unless you had like 12 years worth of data or greater the year embedding probably wouldn't help. As it would require a lot of differences between different years of data to see an impact on the forecasting based on the differences in the representation. We could possibly add it in the future but I doubt most people have a long enough temporal history for it to really help.
